### DISCUSSION TITLE
Rust API for submitting jobs

### DISCUSSION THREAD

**Discussion by apraga** [It4innovations/hyperqueue#763] [Q&A] [answered]
Hi,

Thanks for your package, it looks awesome one the CLI !
I was wondering what would be the best way to submit jobs directly from Rust (like pyhq) without needing to wait on them. 

I would be interested in a minimal working example with a server already started. 

Thanks

**Comment by Kobzol** [comment, marked as answer]
Hi, we don't currently have a Rust API, nor C bindings that could be used from Rust. The easiest way right now would probably be to just spawn commands using the `hq` binary, and using the [JSON output mode](https://it4innovations.github.io/hyperqueue/stable/cli/output-mode/#json) so that you can programmatically read the output.

**Comment by spirali** [comment]
Hi,

I think that the most simple way for the current version would be a compose a TOML file in Rust and call `hq job submit-file`.

https://it4innovations.github.io/hyperqueue/stable/jobs/jobfile/

In longer perspective, we are planning to make some common API for submitting. Mainly because of simplify intermediate layer between HQ and Python, but if there is a usecase it should be posible to also provide it as a Rust crate.

**Comment by apraga** [comment]
Thank you, I’ll run hq as a shell command for the time being :)
